packages/app/src/app/overmind/namespaces/git/actions.ts

```diff
@@ -871,3 +871,19 @@ export const _tryResolveConflict: AsyncAction = async ({
     git.gitState = SandboxGitState.SYNCED;
   }
 };
+
+export const enableGitSync: AsyncAction<string> = async (
+  { state, effects, actions },
+  sandboxId
+) => {
+  if (!state.editor.currentSandbox) return;
+  state.git.isEnablingSync = true;
+  const newGitData = await effects.api.makeGitSandbox(sandboxId);
+  state.editor.currentSandbox = {
+    ...state.editor.currentSandbox,
+    originalGitCommitSha: newGitData.originalGitCommitSha,
+    originalGit: newGitData.originalGit,
+  };
+  state.git.isEnablingSync = false;
```

Comment:
Maybe we should wrap this in a `try...catch` and put `state.git.isEnablingSync = false`  in the `finally`. Not much could go wrong here, but it's a good way to ensure that we can't get stuck.

Reply:
done :)